lib/bitswap.go

```diff
@@ -54,8 +55,12 @@ func (o *BsCheckOutput) MarshalJSON() ([]byte, error) {
 
 var _ json.Marshaler = (*BsCheckOutput)(nil)
 
-func CheckBitswapCID(ctx context.Context, c cid.Cid, ma multiaddr.Multiaddr, getBlock bool) (*BsCheckOutput, error) {
-	h, err := libp2pHost()
+func CheckBitswapCID(ctx context.Context, h host.Host, c cid.Cid, ma multiaddr.Multiaddr, getBlock bool) (*BsCheckOutput, error) {
```

Comment:
If we're going to add in a "you can pass a nil host but then there are defaults" thing there should be a comment for it.

While this seems generally useful I don't think the way it's used in the reason this PR was filed (i.e. https://github.com/ipfs-shipyard/ipfs-check/pull/37/files#r1699032306) is quite right.